File: 1-Two-Sum/solution.py
func twoSum(nums []int, target int) []int {
    indexs := make([]int, 2)

	hash := map[int]int{}

	for i := range nums {
		hash[target-nums[i]] = i
	}

	for i := range nums {
		index, ok := hash[nums[i]]
		if !ok {
			continue
		}

		if index == i {
		    continue
		}
		
		if index > i {
		    indexs[0] = i
		    indexs[1] = index
		} else {
		    indexs[0] = index
		    indexs[1] = i
		}
		
		break
	}
	return indexs
}

File: 1-Two-Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mapping;
        vector<int> res;
        for (int i = 0; i < nums.size(); i++)
        {
            mapping[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++)
        {
            const int gap = target - nums[i];
            if (mapping.find(gap) != mapping.end())
            {
                if (i ^ mapping[gap])
                {
                    res.push_back(i);
                    res.push_back(mapping[gap]);
                    break;
                }
            }
        }
        return res;
    }
};